File: java/testing/TestImageStream.java
import app.f3d.F3D.*;

public class TestImageStream {
  public static void main(String[] args) {
    Image img1 = new Image(300, 200, 3);

    byte[] img1Buffer = img1.saveBuffer(Image.SaveFormat.PNG);

    Image img2 = new Image(img1Buffer);

    img1.equals(img2);

    img1.delete();
    img2.delete();
  }
}

File: python/testing/test_image_stream.py
import os
from pathlib import Path
import tempfile
import pytest

import f3d

@pytest.fixture
def f3d_engine():
    engine = f3d.Engine.create(True)
    engine.window.size = 300, 200
    return engine

def test_loading_stream(f3d_engine: f3d.Engine):
    img = f3d_engine.window.render_to_image(True)
    buffer = img.save_buffer(f3d.Image.SaveFormat.PNG)

    stream_image = f3d.Image(buffer)
    assert stream_image == img
